code/__HELPERS/unsorted.dm

```diff
@@ -508,17 +508,15 @@ Turf and target are separate in case you want to teleport some distance from a t
 /*
 	Gets all contents of contents and returns them all in a list.
 */
-/atom/proc/GetAllContents()
-	var/list/processing_list = list(src)
-	var/list/assembled = list()
-	while(processing_list.len)
-		var/atom/A = processing_list[1]
-		processing_list.Cut(1, 2)
-		//Byond does not allow things to be in multiple contents, or double parent-child hierarchies, so only += is needed
-		//This is also why we don't need to check against assembled as we go along
-		processing_list += A.contents
-		assembled += A
-	return assembled
+
+/atom/proc/GetAllContents(list/output=list(), recursive_depth=INFINITY)
+	. = output
+	output += src
+	if(!recursive_depth)
+		return
+	for(var/i in 1 to contents.len)
+		var/atom/thing = contents[i]
+		thing.GetAllContents(output, recursive_depth-1)
```

Comment:
Why is this reverted?

Reply:
It has reverted my contributor. I can make it back if needed.

Reply:
reverted your contributor?

Reply:
Guy,who helped me with some things.